validator/src/main.rs

```diff
@@ -168,21 +140,48 @@ fn start_gossip_node(
 
 fn get_rpc_node(
     cluster_info: &ClusterInfo,
+    entrypoint_gossip: &SocketAddr,
     validator_config: &ValidatorConfig,
     blacklisted_rpc_nodes: &mut HashSet<Pubkey>,
     snapshot_not_required: bool,
     no_untrusted_rpc: bool,
 ) -> (ContactInfo, Option<(Slot, Hash)>) {
     let mut blacklist_timeout = Instant::now();
     loop {
-        info!(
-            "Searching for an RPC service, shred version={:?}...",
-            validator_config.expected_shred_version
-        );
         sleep(Duration::from_secs(1));
         info!("\n{}", cluster_info.contact_info_trace());
 
-        let rpc_peers = get_shred_rpc_peers(&cluster_info, validator_config.expected_shred_version);
+        let shred_version = validator_config
```

Comment:
If `--expected-shred-version` is not provided, grab our current shred version from cluster info.  This will be set to non-zero shortly after receiving ContactInfo for our entrypoint.

ref: https://github.com/solana-labs/solana/blob/c8f03c7f6de9c334501f2e5541c1a2e501a8f59d/core/src/cluster_info.rs#L1492-L1493